nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/batch/Batch.java

```diff
@@ -210,43 +211,61 @@ public void clean() {
         BatchManager bm = Framework.getService(BatchManager.class);
         TransientStore ts = bm.getTransientStore();
         for (String fileIndex : fileEntries.keySet()) {
-            // Check for chunk entries to remove
-            BatchFileEntry fileEntry = (BatchFileEntry) getFileEntry(fileIndex, false);
-            if (fileEntry != null) {
-                if (fileEntry.isChunked()) {
-                    for (String chunkEntryKey : fileEntry.getChunkEntryKeys()) {
-                        // Remove chunk entry from the store and delete blobs from the file system
-                        List<Blob> chunkBlobs = ts.getBlobs(chunkEntryKey);
-                        if (chunkBlobs != null) {
-                            for (Blob blob : chunkBlobs) {
-                                try {
-                                    FileUtils.deleteDirectory(blob.getFile().getParentFile());
-                                } catch (IOException e) {
-                                    log.error("Error while deleting chunk parent directory", e);
-                                }
+            removeFileEntry(fileIndex, ts);
+        }
+        // Remove batch entry
+        ts.remove(key);
+    }
+
+    /**
+     * @since 8.4
+     */
+    public boolean removeFileEntry(String index, TransientStore ts) {
+        // Check for chunk entries to remove
+        BatchFileEntry fileEntry = getFileEntry(index, false);
+        if (fileEntry != null) {
```

Comment:
I'd go with `if (fileEntry == null) {return false;}` without `else` allowing to remove one indentation level for the successful case.